sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -629,6 +629,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val WHOLESTAGE_CODEGEN_USE_ID_IN_CLASS_NAME =
+    buildConf("spark.sql.codegen.wholeStage.useIdInClassName")
```

Comment:
nit: It seems other whole stage codegen configs don't have the `wholeStage` prefix, shall we remove it to make them consistent?

Reply:
I'm open to suggestions for the config option name. Do you have any concrete suggestions?
Looks like you're suggesting `spark.sql.codegen.useIdInClassName`, right?

I chose the current name (prefix) for two reasons:
1. the config option right before mine is named `spark.sql.codegen.wholeStage`, and I just used it as a prefix
2. this option only affects whole-stage codegen and not other (expression/predicate/ordering/encoder) codegens.

But you're making a good point that all the other whole-stage codegen config options (the ones following this one) only use `spark.sql.codegen` as the prefix. So if you'd confirm that I understood your suggestion correctly, I'll update the PR to address it.

Thanks!

Reply:
Logically it's better to have the `wholeStage` prefix, let's start from here that always adding `wholeStage` prefix to whole stage related config.(keep the existing one unchanged)

cc @viirya